File: Python/5kyu/moving_zeros_to_the_end.py
# Write an algorithm that takes an array and moves all of the zeros to the end, preserving the order of the other elements.

# move_zeros([1, 0, 1, 2, 0, 1, 3]) # returns [1, 1, 2, 1, 3, 0, 0]

def move_zeros(lst):
    #need to move all the zeros to the end
    zeros = lst.count(0)
    nums = [i for i in lst if i != 0]
    for i in range(zeros):
        nums.append(0)
    return nums

# top solution
def move_zeros(arr):
    l = [i for i in arr if isinstance(i, bool) or i!=0]
    return l+[0]*(len(arr)-len(l))

File: JavaScript/5kyu/movingZerosToTheEnd.js
// Write an algorithm that takes an array and moves all of the zeros to the end, preserving the order of the other elements.

// moveZeros([false,1,0,1,2,0,1,3,"a"]) // returns[false,1,1,2,1,3,"a",0,0]

function moveZeros(arr) {
    let result = []
    let zeros_count = 0
    for(let i = 0; i < arr.length; i++){
      if(arr[i] === 0){
        zeros_count += 1
      } else{
        result.push(arr[i])
      }
    }
    let zeroes = Array(zeros_count).fill(0)
    return result.concat(zeroes)
  
  }

// top solution
var moveZeros = function (arr) {
    var filtedList = arr.filter(function (num){return num !== 0;});
    var zeroList = arr.filter(function (num){return num === 0;});
    return filtedList.concat(zeroList);
  }